script/DeployL2OFTAdapter.s.sol

```diff
@@ -0,0 +1,163 @@
+
+import {BaseScript} from "./BaseScript.s.sol";
+
+import {L2YnERC20Upgradeable} from "@/L2YnERC20Upgradeable.sol";
+import {L2YnOFTAdapterUpgradeable} from "@/L2YnOFTAdapterUpgradeable.sol";
+import {ImmutableMultiChainDeployer} from "@/factory/ImmutableMultiChainDeployer.sol";
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {console} from "forge-std/console.sol";
+
+// forge script script/DeployL2OFTAdapter.s.sol:DeployL2Adapter \
+// --rpc-url ${rpc} --sig "run(string calldata)" ${path} \
+// --account ${deployerAccountName} --sender ${deployer} \
+// --broadcast --etherscan-api-key ${api} --verify
+
+contract DeployL2OFTAdapter is BaseScript {
+    ImmutableMultiChainDeployer public multiChainDeployer;
+    L2YnOFTAdapterUpgradeable public l2OFTAdapter;
+    L2YnERC20Upgradeable public l2ERC20;
+
+    function run(string calldata _jsonPath) public {
+        _loadInput(_jsonPath);
+
+        require(currentDeployment.isL1 != true, "Must be L2 deployment");
+
+        bytes32 salt = createImmutableMultiChainDeployerSalt(msg.sender);
+
+        address predictedAddress = predictions.l2MultiChainDeployer;
+        if (!isContract(currentDeployment.multiChainDeployer)) {
+            vm.broadcast();
+            multiChainDeployer = new ImmutableMultiChainDeployer{salt: salt}();
+            console.log("Deployed ImmutableMultiChainDeployer at: ", address(multiChainDeployer));
+        } else {
+            console.log("Already deployed ImmutableMultiChainDeployer at: ", currentDeployment.multiChainDeployer);
+            multiChainDeployer = ImmutableMultiChainDeployer(predictedAddress);
+        }
+
+        require(address(multiChainDeployer) == predictedAddress, "Prediction mismatch");
+
+        currentDeployment.multiChainDeployer = address(multiChainDeployer);
+
+        bytes32 proxySalt = createL2YnERC20UpgradeableProxySalt(msg.sender);
+        bytes32 implementationSalt = createL2YnERC20UpgradeableSalt(msg.sender);
+
+        address deployer = msg.sender;
+
+        address predictedERC20 = multiChainDeployer.getDeployed(proxySalt);
+        require(predictedERC20 == predictions.l2ERC20, "Prediction mismatch");
+
+        if (!isContract(currentDeployment.erc20Address)) {
+            vm.broadcast();
+            l2ERC20 = L2YnERC20Upgradeable(
+                multiChainDeployer.deployL2YnERC20(
+                    implementationSalt,
+                    proxySalt,
+                    baseInput.erc20Name,
+                    baseInput.erc20Symbol,
+                    deployer,
+                    deployer,
+                    type(L2YnERC20Upgradeable).creationCode
+                )
+            );
+
+            vm.broadcast();
+
+            address newOwner = getData(block.chainid).PROXY_ADMIN;
+            console.log("Changing owner for L2ERC20 to: %s", newOwner);
+            Ownable(getTransparentUpgradeableProxyAdminAddress(address(l2ERC20))).transferOwnership(newOwner);
```

Comment:
Owned by a newly deployed timelock

The timelock should be fully owned and controlled by the security council what is now called PROXY_ADMIN in your script.